### PR TITLE
feat: add unit tests for authorization

### LINKED ISSUE #20: Feat: add test coverage
### Feature Description

Add automated test coverage generation. This will probably use Jacoco for the backend. Research what to use for the front-end.

### User Story

_No response_

### Use Cases

_No response_

### Expected Behavior

_No response_

### UI/UX Design (if applicable)

_No response_

### Acceptance Criteria

- [ ] Test coverage can be generated using a maven command on the backend
- [ ] Test coverage is generated after running tests in the CI/CD pipeline
- [ ] Test coverage is shown in open pull-requests

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by A-Bekar**

### CODE REVIEW

File: src/main/java/org/compairifuel/compairifuelapi/fuelprice/presentation/FuelPriceResponseDTO.java [hunk 1]
```diff
@@ -2,12 +2,14 @@
 
 import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.Generated;
 import lombok.RequiredArgsConstructor;
 import org.compairifuel.compairifuelapi.utils.presentation.PositionDTO;
 
 @AllArgsConstructor
 @RequiredArgsConstructor
```
Comment (Mino03947) [inline src/main/java/org/compairifuel/compairifuelapi/fuelprice/presentation/FuelPriceResponseDTO.java]: Volgensmij is deze `@RequiredArgsConstructor` niet nodig. https://projectlombok.org/features/constructor

**Review by Mino03947 — CHANGES_REQUESTED** [review]
For your current draft, i viewed 24 out of 26 files. And left some minor changes and or suggestions.

File: src/test/java/org/compairifuel/compairifuelapi/authorization/service/AuthorizationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    private String grantType = "authorization_code";
+    private String authorizationCode = "authorization_code";
+    private String redirectUri = "https://example.com/callback";
+    private String jwtToken = "JwtToken";
+    private String clientId = "client_id";
+    private String clientSecret = "dummy_secret";
+    private String codeVerifier = "correct_code";
+    private String tokenType = "bearer";
+    private String codeChallenge = "correct_code";
```
Comment (Mino03947) [inline src/test/java/org/compairifuel/compairifuelapi/authorization/service/AuthorizationServiceImplTest.java]: Why are you having a code challenge variable for only the correct edge case, when you use the arrange for R94. 

I don't really prefer having class attributes, this way for testing, but with a good enough reason, i will reconsider.

File: src/test/java/org/compairifuel/compairifuelapi/authorization/service/AuthorizationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    private String refreshToken = "refresh_token";
+    private long expiresIn = 3600000L;
+
+    @BeforeEach
+    void setUp() {
+        closeable = MockitoAnnotations.openMocks(this);
```
Comment (Mino03947) [inline src/test/java/org/compairifuel/compairifuelapi/authorization/service/AuthorizationServiceImplTest.java:42]: What does this do?

Comment (A-Bekar) [reply src/test/java/org/compairifuel/compairifuelapi/authorization/service/AuthorizationServiceImplTest.java:42]: https://www.javadoc.io/doc/org.mockito/mockito-core/3.6.28/org/mockito/MockitoAnnotations.html

Makes the code a bit cleaner IMO.

File: src/main/java/org/compairifuel/compairifuelapi/gasstation/service/GasStationServiceImpl.java [hunk 4]
```diff
@@ -3,13 +3,13 @@
 import jakarta.enterprise.inject.Default;
 import jakarta.inject.Inject;
 import lombok.Data;
```
Comment (Mino03947) [inline src/main/java/org/compairifuel/compairifuelapi/gasstation/service/GasStationServiceImpl.java]: This import is not needed

File: src/main/java/org/compairifuel/compairifuelapi/gasstation/service/GasStationServiceImpl.java [hunk 5]
```diff
@@ -3,13 +3,13 @@
...
+import lombok.Generated;
```
Comment (Mino03947) [inline src/main/java/org/compairifuel/compairifuelapi/gasstation/service/GasStationServiceImpl.java]: This import is also not needed.

**Comment by Mino03947** [comment]
Rebase, and try to solve the sonarcloud issues.

**Review by Mino03947 — APPROVED** [review]
@JaronZ can leave its own opinions, but i reviewed and approve. I did notice a few minor issues, but that might be due to not forcing to use a linter.

**Review by JaronZ — APPROVED** [review]
